synapse/storage/controllers/state.py

```diff
@@ -175,9 +175,9 @@ async def get_state_groups(
 
     @trace
     @tag_args
-    def _get_state_groups_from_groups(
+    async def _get_state_groups_from_groups(
         self, groups: List[int], state_filter: StateFilter
-    ) -> Awaitable[Dict[int, StateMap[str]]]:
```

Comment:
What's the reasoning for why we would choose to return `Awaitable` instead of using `async`?

Reply:
It was probably when we converted from `inlineDeferreds` it would be useful to not add another `Deferred` to the stack? I suspect with async/await support it doesn't really matter TBH.